src/google-maps/google-maps.service.ts

```diff
@@ -1,32 +1,33 @@
-import { Injectable } from '@nestjs/common';
-import { Client } from "@googlemaps/google-maps-services-js";
-import { ConfigService } from "@nestjs/config";
-
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import {
+  Client,
+  Status,
+  TravelMode,
+} from '@googlemaps/google-maps-services-js';
+import { ConfigService } from '@nestjs/config';
 
 @Injectable()
-export class GoogleMapsService extends Client{
+export class GoogleMapsService extends Client {
   private readonly accessKey = this.configService.get('GOOGLE_MAPS_ACCESS_KEY');
   constructor(private configService: ConfigService) {
     super();
   }
 
-  async getCoordinates (address: {
-    origins
-    destinations
-    mode
-  }){
+  async getCoordinates(address: {
+    origins: string[];
+    destinations: string[];
+    mode: TravelMode;
+  }): Promise<number | undefined> {
     const googleRes = await this.distancematrix({
       params: {
         ...address,
-        key:this.accessKey,
-
-      }
-    })
-    const distance = googleRes?.data?.rows[0]?.elements[0]?.distance?.value
-    return distance
-
+        key: this.accessKey,
+      },
+    });
+    if (googleRes.data.rows[0].elements[0].status !== Status.OK) {
```

Comment:
it is worth doing optional chaining here

Reply:
It's done